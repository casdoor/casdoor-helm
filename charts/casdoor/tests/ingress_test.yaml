suite: test ingress
templates:
- ingress.yaml

tests:
- it: should not render when disabled
  asserts:
  - hasDocuments:
      count: 0

- it: should render with default values when enabled
  set:
    ingress.enabled: true
  asserts:
  - matchSnapshot: {}

- it: should set ingress className
  set:
    ingress.enabled: true
    ingress.className: nginx
  asserts:
  - matchSnapshot: {}

- it: should set custom host and path
  set:
    ingress.enabled: true
    ingress.hosts:
    - host: casdoor.example.com
      paths:
      - path: /casdoor
        pathType: Prefix
  asserts:
  - matchSnapshot: {}

- it: should add tls configuration
  set:
    ingress.enabled: true
    ingress.tls:
    - secretName: casdoor-tls
      hosts:
      - casdoor.example.com
  asserts:
  - matchSnapshot: {}
