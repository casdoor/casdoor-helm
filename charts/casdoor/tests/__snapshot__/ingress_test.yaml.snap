should add tls configuration:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-casdoor
                    port:
                      number: 8000
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - casdoor.example.com
          secretName: casdoor-tls
should render with default values when enabled:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-casdoor
                    port:
                      number: 8000
                path: /
                pathType: ImplementationSpecific
should set custom host and path:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      rules:
        - host: casdoor.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-casdoor
                    port:
                      number: 8000
                path: /casdoor
                pathType: Prefix
should set ingress className:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: casdoor
        app.kubernetes.io/version: v1.976.0
        helm.sh/chart: casdoor-helm-charts-v1.976.0
      name: RELEASE-NAME-casdoor
    spec:
      ingressClassName: nginx
      rules:
        - host: chart-example.local
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-casdoor
                    port:
                      number: 8000
                path: /
                pathType: ImplementationSpecific
